### ISSUE TITLE
Defining Optimizer Objects and single Train Step

### ISSUE THREAD

**Issue by Atharva-Gundawar** [ACM-VIT/Fill-In-the-Blanks#11]
**You have to define the Training steps and it's required objects**

The Training step should follow the following guidelines :

Inputs: (some/all of) Original Image, Mask, Image with Mask Applied, Original Content of the masked area

Steps:
- Pass input through the models
- Find Adverserial Loss (as Some linear combination of Generator loss, Global Discriminator Loss, Local Discriminator Loss)
- Apply Loss to the models using the above Optimizer Objects
